core/templates/pages/exploration-editor-page/improvements-tab/improvements-tab.component.ts

```diff
@@ -13,109 +13,159 @@
-          } else if (this.getNumTasks() > 0) {
-            return ImprovementsConstants.EXPLORATION_HEALTH_TYPE_WARNING;
-          } else {
-            return ImprovementsConstants.EXPLORATION_HEALTH_TYPE_HEALTHY;
-          }
-        };
-        this.getNumCardLevelTasksForState = (stateName: string) => {
-          const stateTasks = (
-            ExplorationImprovementsTaskRegistryService.getStateTasks(
-              stateName));
-          const ngrTask = stateTasks.ngrTask;
-          const siaTask = stateTasks.siaTask;
-          return (ngrTask.isOpen() ? 1 : 0) + (siaTask.isOpen() ? 1 : 0);
-        };
-
-        this.allStateTasks = (
-          ExplorationImprovementsTaskRegistryService.getAllStateTasks());
-        const namesOfStatesWithTasks = (
-          this.allStateTasks.map(stateTasks => stateTasks.stateName));
-
-        const stateRetentions: Map<string, number> = (
-          new Map(namesOfStatesWithTasks.map(stateName => {
-            const stateStats = expStats.getStateStats(stateName);
-            const numCompletions = stateStats.numCompletions;
-            const totalHitCount = stateStats.totalHitCount;
-            const retentionRate = Math.round(
-              totalHitCount ? (100.0 * numCompletions / totalHitCount) : 0);
-            return [stateName, retentionRate];
-          })));
-
-        this.getStateRetentionPercent = (stateName: string) => (
-          stateRetentions.get(stateName) + '%');
-
-        const stateVisibility: Map<string, boolean> = (
-          new Map(namesOfStatesWithTasks.map(stateName => [stateName, true])));
-
-        this.isStateTasksVisible = (stateName: string) => (
-          stateVisibility.get(stateName));
-        this.toggleStateTasks = (stateName: string) => (
-          stateVisibility.set(stateName, !stateVisibility.get(stateName)));
-      };
-    },
-  ],
-});
+import { Component, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HighBounceRateTask } from 'domain/improvements/high-bounce-rate-task.model';
+import { ImprovementsConstants } from 'domain/improvements/improvements.constants';
+import { IneffectiveFeedbackLoopTask } from 'domain/improvements/ineffective-feedback-loop-task.model';
+import { NeedsGuidingResponsesTask } from 'domain/improvements/needs-guiding-response-task.model';
+import { SuccessiveIncorrectAnswersTask } from 'domain/improvements/successive-incorrect-answers-task.model';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ExplorationImprovementsTaskRegistryService, StateTasks } from 'services/exploration-improvements-task-registry.service';
+import { RouterService } from '../services/router.service';
+
+type HbrTask = HighBounceRateTask;
+type IflTask = IneffectiveFeedbackLoopTask;
+type NgrTask = NeedsGuidingResponsesTask;
+type SiaTask = SuccessiveIncorrectAnswersTask;
```

Comment:
Why is this needed? IMO it just makes the code more confusing.

Reply:
removed